bring/src/test/java/com/bobocode/petros/bring/factory/DefaultBeanFactoryTest.java

```diff
@@ -0,0 +1,148 @@
+package com.bobocode.petros.bring.factory;
+
+import com.bobocode.petros.bring.context.domain.BeanDefinition;
+import com.bobocode.petros.bring.context.domain.BeanReference;
+import com.bobocode.petros.bring.context.domain.BeanScope;
+import com.bobocode.petros.bring.exception.NoSuchBeanDefinitionException;
+import com.bobocode.petros.bring.factory.mocks.EveningService;
+import com.bobocode.petros.bring.factory.mocks.GreetingService;
+import com.bobocode.petros.bring.factory.mocks.MorningService;
+import com.bobocode.petros.bring.registry.BeanDefinitionRegistry;
+import com.bobocode.petros.bring.registry.DefaultBeanDefinitionRegistry;
+import com.bobocode.petros.bring.registry.utils.RegistryTestUtils;
+import com.bobocode.petros.bring.utils.BeanTestUtils;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.Spy;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Objects;
+import java.util.concurrent.ConcurrentHashMap;
+
+import static org.junit.jupiter.api.Assertions.*;
+import static org.mockito.MockitoAnnotations.openMocks;
+
+@ExtendWith(MockitoExtension.class)
+class DefaultBeanFactoryTest {
+    private BeanFactory factory;
+    private BeanDefinitionRegistry registry;
+    @Spy
+    private Map<String, BeanDefinition> beanDefinitions = new ConcurrentHashMap<>();
+
+    @BeforeEach
+    public void setup() {
+        openMocks(this);
+        registry = DefaultBeanDefinitionRegistry.getInstance();
+        RegistryTestUtils.configureRegistry(registry, beanDefinitions);
+        factory = new DefaultBeanFactory(registry);
+    }
+
+    @AfterEach
+    public void cleanup() {
+        this.beanDefinitions.clear();
+    }
+
+    @Test
+    void whenGetAllBeanReferenceInvokeWithNonEmptyBeanDefinitionRegistryThenReturnMap() {
+        //given
+        String firstBeanName = "morningService";
+        BeanDefinition firstBeanDefinition = BeanTestUtils.createBeanDefinition(firstBeanName, MorningService.class);
+
+        String secondBeanName = "eveningService";
+        BeanDefinition secondBeanDefinition = BeanTestUtils.createBeanDefinition(secondBeanName, EveningService.class);
+
+        registry.registerBeanDefinition(firstBeanName, firstBeanDefinition);
+        registry.registerBeanDefinition(secondBeanName, secondBeanDefinition);
+
+        //then
+        Map<String, BeanReference> referenceMap = factory.getAllBeanReferences();
+
+        // assertions & verification
+        assertEquals(2, referenceMap.size());
```

Comment:
MINOR(FYI). for multiple assertions it's better to use assertAll() method that accepts multiple assertions, so you will be aware of all failed assertions at the end of test execution. And with such approach, you'll get only first fail